user/tlu/cmake/FindTLUFirmware.cmake

```diff
@@ -1,42 +1,49 @@
-  ${CMAKE_CURRENT_LIST_DIR}/../extern/*/TLU_Toplevel.bit)
+include(FetchContent)
 
-# could not find the package at the usual locations -- try to copy from AFS if accessible
-if (NOT EUDET_TLU_FIRMWARE_FOUND_LIST)
-  if (EXISTS "/afs/desy.de/group/telescopes/tlu/tlufirmware")
-    message(STATUS "Could not find tlu firmware package required by tlu producer; downloading it now via AFS to ${CMAKE_CURRENT_LIST_DIR}/../extern  ....")
-    if(DEFINED ENV{TRAVIS})
-      message(STATUS "Running on travis and therefore downloading only the absolutely necessary part of the tlu firmware.")
-      file(MAKE_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}/../extern/tlufirmware)
-      copy_files("/afs/desy.de/group/telescopes/tlu/tlufirmware/TLU_Toplevel.bit" ${CMAKE_CURRENT_LIST_DIR}/../extern/tlufirmware)
-    else()
-      copy_files("/afs/desy.de/group/telescopes/tlu/tlufirmware" ${CMAKE_CURRENT_LIST_DIR}/../extern)
-    endif()
-    find_zestsc1_in_extern(NO_DEFAULT_PATH)
-  else()
-      message(STATUS "Could not find tlu firmware package required by tlu producer. Please refer to the documentation on how to obtain the software.")
-      return()
-  endif()
-  set(EUDET_TLU_FIRMWARE_FILE ${CMAKE_CURRENT_LIST_DIR}/../extern/tlufirmware)
-  get_filename_component(EUDET_TLU_FIRMWARE_FILE ${EUDET_TLU_FIRMWARE_FILE} ABSOLUTE)
-  get_filename_component(EUDET_TLU_FIRMWARE_DIR ${EUDET_TLU_FIRMWARE_FILE} DIRECTORY)
-else()
-  list(GET EUDET_TLU_FIRMWARE_FOUND_LIST 0 EUDET_TLU_FIRMWARE_FILE)
-  get_filename_component(EUDET_TLU_FIRMWARE_FILE ${EUDET_TLU_FIRMWARE_FILE} ABSOLUTE)
-  get_filename_component(EUDET_TLU_FIRMWARE_DIR ${EUDET_TLU_FIRMWARE_FILE} DIRECTORY)
-endif()
+macro(find_bitfiles_in_extern arg)
+  find_path(TLUFIRMWARE_PATH TLU_Toplevel.bit
+    HINTS ${PROJECT_SOURCE_DIR}/extern/tlufirmware ${arg})
+endmacro()
+
+find_bitfiles_in_extern("")
 
-if(EUDET_TLU_FIRMWARE_DIR)
-  set(TLUFIRMWARE_PATH ${EUDET_TLU_FIRMWARE_DIR})
-  set(TLUFIRMWARE_DEFINITIONS "-DTLUFIRMWARE_PATH=\"${EUDET_TLU_FIRMWARE_DIR}\"" )
-  message(STATUS "Check for EUDET_TLU_FIRMWARE_DIR(TLU_Toplevel.bit): ${EUDET_TLU_FIRMWARE_DIR} -- ok")
-else()
-  message(WARNING "Check for EUDET_TLU_FIRMWARE_DIR (TLU_Toplevel.bit) -- false")
+# could not find the package at the usual locations -- it will try to copy from github
+if (NOT TLUFIRMWARE_PATH)
+#  IF (EXISTS "/afs/desy.de/group/telescopes/tlu/tlufirmware")
+#    MESSAGE(STATUS "Could not find TLUFirmware bitfiles required by tlu producer; downloading them now via AFS to ./extern ....")
+#    FILE(COPY "/afs/desy.de/group/telescopes/tlu/tlufirmware" DESTINATION ${PROJECT_SOURCE_DIR}/extern)
+#    find_bitfiles_in_extern(NO_DEFAULT_PATH)
+#  ELSE()
+#    MESSAGE(WARNING "Could not find TLUFirmware bitfiles required by tlu producer. Please refer to the documentation on how to obtain them.  Converter will be built anyway.")
+#  ENDIF()
+  FetchContent_Declare(
+    tlu-dependencies
+    GIT_REPOSITORY    https://github.com/eudaq/tlu-dependencies.git
```

Comment:
Nobody seems to have cared about this, but the reason for placing it on AFS rather than uploading to GitHub is that these are commercial binary drivers that are not supposed to be redistributed by third parties IIRC.

Reply:
Anything else that you want to discuss in a separate MR?